prometheus.go

```diff
@@ -14,20 +14,37 @@ func (g *Gaz) InitPrometheus(path string) {
 	if !strings.HasPrefix(path, "/") {
 		path = "/" + path
 	}
+	g.prometheusRegistry = prometheus.NewRegistry()
 	Sugar.Infof("Setup Prometheus handler at %s", path)
 	g.Router.Handle(path, promhttp.Handler()).Methods("GET")
 
 	// export uptime as a prometheus counter
-	upCounter := promauto.NewCounter(prometheus.CounterOpts{
+	upCounter := prometheus.NewCounter(prometheus.CounterOpts{
 		Name: "uptime_sec",
 		Help: "uptime in seconds",
 	})
+	if g.RegisterCollector(upCounter) {
+		go func() {
+			t := time.Tick(time.Second)
+			for {
+				<-t
+				upCounter.Inc()
+			}
+		}()
+	}
+}
 
-	go func() {
-		t := time.Tick(time.Second)
-		for {
-			<-t
-			upCounter.Inc()
+// return true if collector was successfully registered
+func (g *Gaz) RegisterCollector(c prometheus.Collector) bool {
+	if g.prometheusRegistry != nil {
```

Comment:
Does it make sense to have the possibility to have a nil prometheusRegistry? Shouldn't we always create one?
This will make the code much simpler, not having to check all the time if RegisterCollector(stuff)  {}

Reply:
Ok, we can always create a registry, but the Collector registration can return an error, which is most likely pointing to a programming mistake (duplicate names, etc.)